### PR TITLE
API can attach a Broker to 'started' inquiries

### PR DESCRIPTION
**Pull request by oliverochman**
PT https://www.pivotaltracker.com/story/show/178449305

### User Story
```
As a Broker
In order to have a better overview of who is doing what inquiry
I'd like to attach a Broker to an Inquiry
```
### Changes proposed 
- When we do an `update` action, we associate a Broker with with an `inquiry` 
- Add permissions for who can update an inquiry from "started" to "done"
- Restructure of req spec file for Inquires#update action
- Add User model with devise token auth
### Packages/Gems added
- gem 'pundit'
- gem 'devise-token-auth'
- gem 'pundit_helpers'

### What I have learned working on this feature
- How to use` pundit` & `pundit helpers`

### CODE REVIEW

File: app/policies/inquiry_policy.rb [hunk 1]
```diff
@@ -0,0 +1,10 @@
+class InquiryPolicy < ApplicationPolicy
+  # class Scope < Scope
```
Comment (oliverochman) [inline app/policies/inquiry_policy.rb]: remove commented out code

File: config/initializers/devise_token_auth.rb [hunk 2]
```diff
@@ -0,0 +1,60 @@
+# frozen_string_literal: true
+
+DeviseTokenAuth.setup do |config|
+  # By default the authorization headers will change after each request. The
+  # client is responsible for keeping track of the changing tokens. Change
+  # this to false to prevent the Authorization header from changing after
```
Comment (oliverochman) [inline config/initializers/devise_token_auth.rb]: remove commented out code

File: app/controllers/api/inquiries_controller.rb [hunk 3]
```diff
@@ -17,14 +18,33 @@ def index
...
   def update
     inquiry = Inquiry.find(params[:id])
-    inquiry.update(inquiry_status: params[:form_data][:inquiry_status])
-    render json: { inquiry: inquiry, message: 'Inquiry has been updated' }, status: 200
+    if inquiry.pending?
+      inquiry.update(
+        inquiry_status: params[:form_data][:inquiry_status],
+        broker: current_user
+      )
+    else
```
Comment (LucasKnudsen) [inline app/controllers/api/inquiries_controller.rb:26]: If a broker sends `done` with params, won't this update it straight to being done?

File: spec/requests/api/provider_can_update_inquiry_spec.rb [hunk 4]
```diff
@@ -1,40 +1,125 @@
 RSpec.describe 'PUT /api/inquiries/:id', type: :request do
```
Comment (LucasKnudsen) [inline spec/requests/api/provider_can_update_inquiry_spec.rb:1]: Lots of lovely testing - but what about the path from pending to done? (same as above)
Or are we not touching upon this functionality

File: db/migrate/20210609125637_devise_token_auth_create_users.rb [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+      # t.integer  :failed_attempts, :default => 0, :null => false # Only if lock strategy is :failed_attempts
+      # t.string   :unlock_token # Only if unlock strategy is :email or :both
+      # t.datetime :locked_at
+
+      ## User Info
+      t.string :name
+      t.string :nickname
+      t.string :image
+      t.string :email
+
```
Comment (LucasKnudsen) [inline db/migrate/20210609125637_devise_token_auth_create_users.rb]: Do we plan to use all these attributes? Usually we only have email, but I guess we'd also like name for this one.

File: spec/models/user_spec.rb [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+
+  describe "Database table" do
+    it { is_expected.to have_db_column :encrypted_password }
+    it { is_expected.to have_db_column :email }
+  end
+
+  describe "Validations" do
+    it { is_expected.to validate_presence_of :email }
+    it { is_expected.to validate_presence_of :password }
+  end
```
Comment (LucasKnudsen) [inline spec/models/user_spec.rb:14]: Is devise validating presence of those other attributes as well, do we know?

File: app/controllers/api/inquiries_controller.rb (hunk 3 above)
Comment (oliverochman) [reply app/controllers/api/inquiries_controller.rb:26]: In order to fix that we need to make use of a gem, so we will do that in a chore. Quite a lot of logic will be added, and this feature is already big

File: db/migrate/20210609125637_devise_token_auth_create_users.rb (hunk 5 above)
Comment (oliverochman) [reply db/migrate/20210609125637_devise_token_auth_create_users.rb]: fixed

File: spec/models/user_spec.rb (hunk 6 above)
Comment (oliverochman) [reply spec/models/user_spec.rb:14]: We know that devise is working at this point, so we dont have to test every column that it is validating

**Review by arunbhalli — APPROVED** [review]

**Review by lukeperrin10 — APPROVED** [review]

**Review by LucasKnudsen — APPROVED** [review]

**Pull request merged**